contracts/test/unit/pcv/PCVGuardian.t.sol

```diff
@@ -0,0 +1,403 @@
+    }
+
+    function testWithdrawToSafeAddressFailWhenNotWhitelist() public {
+        vm.prank(addresses.governorAddress);
+        vm.expectRevert(
+            bytes("PCVGuardian: Provided address is not whitelisted")
+        );
+
+        pcvGuardian.withdrawToSafeAddress(address(0x1), mintAmount);
+    }
+
+    function testWithdrawAllToSafeAddress() public {
+        vm.startPrank(addresses.governorAddress);
+        assertEq(underlyingToken.balanceOf(address(this)), 0);
+
+        uint256 amountToWithdraw = pcvDeposit.balance();
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+
+        assertEq(underlyingToken.balanceOf(address(this)), amountToWithdraw);
+    }
+
+    function testPCVGuardWithdrawAllToSafeAddress() public {
+        vm.startPrank(guard);
+        assertEq(underlyingToken.balanceOf(address(this)), 0);
+
+        uint256 amountToWithdraw = pcvDeposit.balance();
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+
+        assertEq(underlyingToken.balanceOf(address(this)), amountToWithdraw);
+    }
+
+    function testGuardianWithdrawAllToSafeAddress() public {
+        vm.startPrank(addresses.guardianAddress);
+        assertEq(underlyingToken.balanceOf(address(this)), 0);
+
+        uint256 amountToWithdraw = pcvDeposit.balance();
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+
+        assertEq(underlyingToken.balanceOf(address(this)), amountToWithdraw);
+    }
+
+    function testWithdrawAllToSafeAddressFailWhenNoRole() public {
+        vm.expectRevert(bytes("UNAUTHORIZED"));
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+    }
+
+    function testWithdrawAllToSafeAddressFailWhenGuardRevokedGovernor() public {
+        vm.prank(addresses.governorAddress);
+        pcvGuardAdmin.revokePCVGuardRole(guard);
+
+        vm.prank(guard);
+        vm.expectRevert(bytes("UNAUTHORIZED"));
+
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+    }
+
+    function testWithdrawAllToSafeAddressFailWhenGuardRevokedGuardian() public {
+        vm.prank(addresses.guardianAddress);
+        pcvGuardAdmin.revokePCVGuardRole(guard);
+
+        vm.prank(guard);
+        vm.expectRevert(bytes("UNAUTHORIZED"));
+
+        pcvGuardian.withdrawAllToSafeAddress(address(pcvDeposit));
+    }
+
+    function testWithdrawAlloSafeAddressFailWhenNotWhitelist() public {
+        vm.prank(addresses.governorAddress);
+        vm.expectRevert(
+            bytes("PCVGuardian: Provided address is not whitelisted")
+        );
+
+        pcvGuardian.withdrawAllToSafeAddress(address(0x1));
+    }
+
+    function testGovernorWithdrawERC20ToSafeAddress() public {
+        vm.startPrank(addresses.governorAddress);
+        assertEq(rewardToken.balanceOf(address(this)), 0);
+
+        pcvGuardian.withdrawERC20ToSafeAddress(
+            address(pcvDeposit),
+            address(rewardToken),
```

Comment:
Doesn't matter, but reading the test is a bit weird - you're withdrawing to a reward token. 

I'd probs define a `address user = address(0x1)`, withdraw to that and verify that balance. Functionally the same, but a bit more readable

Reply:
We're withdrawing to the safeAddress, but because we've already decided to only use one we don't need to specify it in the function, unlike the fei version where it does require the safe address to be specified, I think thats where the confusion stems from, if I'm understanding your comment correctly

Reply:
This version of the guardian does not allow you to withdraw to a guardian or governor specified address, only the address specified as the safe address during construction.